src/nitypes/xy_data.py

```diff
@@ -0,0 +1,229 @@
+"""XYData type for NI Python APIs.
+
+XYData Data Type
+=================
+:class:`XYData`: An XYData object represents a two dimensional sequence of numeric values with
```

Comment:
Why XYData is a 2D sequence of numeric values, it's better described as 2xn array. We have 2 arrays, each with n values. 2D suggests arbitrary size of the array.